pkg/api/steve/proxy/proxy.go

```diff
@@ -146,11 +147,27 @@ func (h *Handler) authLocalCluster(router *gmux.Router) func(rw http.ResponseWri
 func (h *Handler) matchManagementCRDs() gmux.MatcherFunc {
 	return func(req *http.Request, match *gmux.RouteMatch) bool {
 		splitPath := strings.Split(req.URL.Path, "/")
-		if len(splitPath) < 3 {
+		var resource string
+		// Extract either a cluster-wide or namespaced resource from the URL.
+		// The matcher function doesn't match paths like /api/v1/pods, which represent core Kubernetes resources.
+		// It also doesn't match those exempt from further authentication - only management resources and
+		// only those found in a special allow-list. The following structure is thus expected for these resources' paths.
+		switch len(splitPath) {
+		case 5: // Management resource at the cluster level, such as /apis/management.cattle.io/v3/podsecurityadmissionconfigurationtemplates
+			resource = splitPath[4]
+		case 7: // Management resource in a namespace, such as /apis/management.cattle.io/v3/namespaces/default/projectroletemplatebindings
+			resource = splitPath[6]
+		default:
 			return false
 		}
 		apiGroup := splitPath[2]
-		return apiGroup == managementv3.GroupName
+		if apiGroup == managementv3.GroupName {
+			if apiaccess.AllowAll[resource] {
+				return false
+			}
+			return true
+		}
+		return false
```

Comment:
This can be one if statement.
```suggestion
		if apiGroup == managementv3.GroupName && !apiaccess.AllowAll[resource]{ 
			return true
		}
		return false
```